programs/drift_vaults/src/state/vault.rs

```diff
@@ -504,4 +732,390 @@ impl Vault {
+            Some(VaultDepositorRecordProtocolParams {
+                protocol_profit_share: 0,
+                protocol_fee: protocol_fee_payment,
+                protocol_fee_shares,
+            }),
+        )?;
+
+        if let Some(vp) = vault_protocol {
+            self.total_withdraw_requested = self
+                .total_withdraw_requested
+                .safe_sub(vp.last_protocol_withdraw_request.value)?;
+            vp.last_protocol_withdraw_request.reset(now)?;
+        }
+
+        Ok(n_tokens)
+    }
+
+    pub fn validate_vault_protocol(&self, vp: &Option<AccountLoader<VaultProtocol>>) -> Result<()> {
+        match vp {
+            None => {
+                if self.vault_protocol {
+                    // Vault has VaultProtocol but no rem acct provided.
+                    let ec = ErrorCode::VaultProtocolMissing;
+                    msg!("Error {} thrown at {}:{}", ec, file!(), line!());
+                    msg!("VaultProtocol missing in remaining accounts");
+                    Err(anchor_lang::error::Error::from(ec))
+                } else {
+                    // Vault does not have VaultProtocol and none given in rem accts.
+                    Ok(())
+                }
+            }
+            Some(vp) => {
+                if self.vault_protocol {
+                    // Vault has VaultProtocol and rem accts provided one.
+                    // check if PDA matches rem acct given.
+                    let (expected, _) = Pubkey::find_program_address(
+                        &[b"vault_protocol", self.pubkey.as_ref()],
+                        &crate::id(),
+                    );
+                    let actual = vp.to_account_info().key();
+                    if actual != expected {
+                        Err(
+                            anchor_lang::error::Error::from(error::ErrorCode::ConstraintSeeds)
+                                .with_account_name("vault_protocol")
+                                .with_pubkeys((actual, expected)),
+                        )
+                    } else {
+                        Ok(())
+                    }
+                } else {
+                    // Vault does not have VaultProtocol, but rem accts provided one
+                    let ec = ErrorCode::VaultProtocolMissing;
+                    msg!("Error {} thrown at {}:{}", ec, file!(), line!());
+                    msg!("Vault does not have VaultProtocol");
+                    Err(anchor_lang::error::Error::from(ec))
+                }
+            }
+        }
+    }
+
+    fn emit_vault_depositor_record(
+        &self,
+        params: VaultDepositorRecordParams,
+        optional_params: Option<VaultDepositorRecordProtocolParams>,
+    ) -> Result<()> {
+        match optional_params {
+            None => {
+                emit!(VaultDepositorRecord {
+                    ts: params.ts,
+                    vault: self.pubkey,
+                    depositor_authority: self.manager,
```

Comment:
for protocol related operations, should this be the protocol or protocol signer?

Reply:
The manager is still the auth for vault ops, including depositor actions.
The protocol is just for collecting a fee, nor does it have any perms over depositors... so I'd say no. 
Do you see something I don't?